test.js

```diff
@@ -46,6 +46,12 @@ test('unist-util-map', function (t) {
     'should work when passing an empty object'
   )
 
+  t.deepEqual(
+    map(u('root', [u('node', [u('leaf', '1')]), u('leaf', '2')]), asIs),
+    u('root', [u('node', [u('leaf', '1')]), u('leaf', '2')]),
+    'should keep node as-is'
+  )
+
```

Comment:
```suggestion
  /** @type {Root} */
  const tree = u('root', [u('node', [u('leaf', '1')]), u('leaf', '2')])

  t.deepEqual(
    map(tree, asIs),
    tree,
    'should support an explicitly typed `MapFunction`'
  )

```

1. By passing an explicit `Root`, TS knows that it’s explicitly that type, otherwise it has to “fit” an object literal into that shape later.
2. Improved message